src/error.h

```diff
@@ -73,6 +73,13 @@ class ValidationError : public Error {
   ValidationError(std::string msg) : Error(msg) {}
 };
 
+/// For depricating API until a next major release
+class DepricatedApiError : public Error {
```

Comment:
Should this name be consistent with  the deprecation warning (sans api) below.  That is, this could be `DeprecationError`